qcodes/dataset/plotting.py

```diff
@@ -552,3 +553,119 @@ def _is_string_valued_array(values: np.ndarray) -> bool:
         True, if the array contains string; False otherwise
     """
     return isinstance(values[0], str)
+
+#
+# Post analysis functions for plot_by_id
+#
+
+
+def _get_axis(coords: np.ndarray, avg_dim: str) -> np.ndarray:
+    """
+    Helper function for _average_heatmap to get the points of either
+    the x- or y-axis from a meshgrid's coordinates
+    """
+    if avg_dim == 'column':
+        bins = coords[0][:, 0]
+    elif avg_dim == 'row':
+        bins = coords[:, 0][:, 1]
+    else:
+        raise ValueError(f'Unknown avg_dim: {avg_dim}')
```

Comment:
newline for readability
```suggestion
        raise ValueError(f'Unknown avg_dim: {avg_dim}')
        
```